### PR TITLE
Enabling test config xml

### PR DESCRIPTION
**Pull request by vilkg**
DefaultDhisConfigurationProvider is registered as a bean only when spring profile is not test. This enables:
- easier test properties configuration,
- having multiple configuration files for different kinds of tests (unit,integration,..)
- removal of if (testing) clauses in DefaultHibernateConfigurationProvider and DefaultDhisConfiguration provider.

### CODE REVIEW

File: dhis-2/dhis-support/dhis-support-test/src/main/java/org/hisp/dhis/config/UnitTestConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package org.hisp.dhis.config;
+
```
Comment (larshelge) [inline dhis-2/dhis-support/dhis-support-test/src/main/java/org/hisp/dhis/config/UnitTestConfiguration.java]: Needs source code license and @author info.

File: dhis-2/dhis-support/dhis-support-test/src/main/java/org/hisp/dhis/external/conf/TestDhisConfigurationProvider.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
+
+package org.hisp.dhis.external.conf;
+
```
Comment (larshelge) [inline dhis-2/dhis-support/dhis-support-test/src/main/java/org/hisp/dhis/external/conf/TestDhisConfigurationProvider.java]: Needs source code license and @author info.

File: dhis-2/dhis-support/dhis-support-test/src/main/java/org/hisp/dhis/DhisTest.java [hunk 3]
```diff
@@ -51,7 +53,8 @@
  * @author Lars Helge Overland
  */
 @RunWith( SpringRunner.class )
-@ContextConfiguration( locations = { "classpath*:/META-INF/dhis/beans.xml" } )
+@ContextConfiguration( classes = { UnitTestConfiguration.class } )
+@ActiveProfiles("test")
```
Comment (ameenhere) [inline dhis-2/dhis-support/dhis-support-test/src/main/java/org/hisp/dhis/DhisTest.java]: Can @ActiveProfiles("test") be moved to DhisConvenienceTest. If its possible, then it wouldnt be required to annotate neither of DhisTest or DhisSpringTest.

File: dhis-2/dhis-support/dhis-support-external/src/main/resources/META-INF/dhis/beans.xml [hunk 4]
```diff
@@ -23,4 +18,9 @@
   <bean id="initialInterval" class="org.hisp.dhis.external.conf.ConfigurationPropertyFactoryBean"><constructor-arg value="META_DATA_SYNC_RETRY_TIME_FREQUENCY_MILLISEC" /></bean>
   <bean id="sessionTimeout" class="org.hisp.dhis.external.conf.ConfigurationPropertyFactoryBean"><constructor-arg value="SYSTEM_SESSION_TIMEOUT" /></bean>
 
+  <beans profile="!test">
+    <bean id="dhisConfigurationProvider" class="org.hisp.dhis.external.conf.DefaultDhisConfigurationProvider" init-method="init">
+      <property name="locationManager" ref="locationManager" />
+    </bean>
+  </beans>
```
[the remark this answers is not in the document]
Comment (vilkg) [reply dhis-2/dhis-support/dhis-support-external/src/main/resources/META-INF/dhis/beans.xml]: We will need another configuration class, where we will spin up docker instance of postgres for integration tests, that's why I left possibility to provide DhisConfigurationProvider from base test class

File: dhis-2/dhis-support/dhis-support-test/src/main/java/org/hisp/dhis/DhisTest.java (hunk 3 above)
Comment (vilkg) [reply dhis-2/dhis-support/dhis-support-test/src/main/java/org/hisp/dhis/DhisTest.java]: Yap, that makes sense!

File: dhis-2/dhis-support/dhis-support-test/src/main/java/org/hisp/dhis/config/UnitTestConfiguration.java (hunk 1 above)
Comment (vilkg) [reply dhis-2/dhis-support/dhis-support-test/src/main/java/org/hisp/dhis/config/UnitTestConfiguration.java]: done

File: dhis-2/dhis-support/dhis-support-test/src/main/java/org/hisp/dhis/external/conf/TestDhisConfigurationProvider.java (hunk 2 above)
Comment (vilkg) [reply dhis-2/dhis-support/dhis-support-test/src/main/java/org/hisp/dhis/external/conf/TestDhisConfigurationProvider.java]: done

File: dhis-2/dhis-support/dhis-support-external/src/main/resources/META-INF/dhis/beans.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (ameenhere) [reply dhis-2/dhis-support/dhis-support-external/src/main/resources/META-INF/dhis/beans.xml]: Ah. Thats great then.

**Pull request merged**